app/models/barcode.rb

```diff
@@ -11,9 +11,18 @@ class Barcode < ApplicationRecord
   require 'sanger_barcode_format/legacy_methods'
   extend SBCF::LegacyMethods
 
+  # This association will have to be removed in a subsequent story
   belongs_to :asset, optional: false, class_name: 'Labware'
+  # New association introduced in #4121. This is a temporary measure to allow the renaming
+  # of the asset association to labware. This was created to accommodate for_search_query scope's include() method
+  belongs_to :labware, class_name: 'Labware', optional: false
```

Comment:
OK so you've actually added a new association - could this be sufficient? Is it necessary to also have `alias_association` and `alias_attribute`?

Reply:
Is it possible to do something like `belongs_to :labware, foreign_key: :asset_id, optional: false` ?

Reply:
Hi Katy, so the `alias_association` adds getters and setters to the `labware` association that wraps the `asset` association. That is, when it calls `barcode.labware` it invokes the getter and when it calls `barcode.labware = ` it invokes the setter. This is how the code references that uses the `asset` association was made interchangeable with `labware` association.

So, I think `alias_association` and `alias_attribute` are necessary to update the code references that uses `asset` to `labware`.

Reply:
And unfortunately, `belongs_to :labware, foreign_key: :asset_id, optional: false` doesn't really work because it fails the validations when building the model: https://github.com/sanger/sequencescape/actions/runs/9660235857/job/26645343790

Reply:
OK, thanks for trying it and the explanations!

I'm still curious as to 
a) why those validations are failing - you'd think that `asset_id` would have to be populated in the factories due to the existing association `belongs_to :asset, optional: false, class_name: 'Labware'`
b) whether we could add `belongs_to :labware, foreign_key: :asset_id` (without the `optional: false` that is presumably causing the validations to fail) - only useful if we could then get rid of the `alias_attribute` and `alias_association` to simplify things.

Feel free to merge though if you feel like you've spent enough time on it!

Reply:
Hi @KatyTaylor, so as far as I've found out, `belongs_to` association adds methods to get the associated object, and assign the associated object (see the image below; it's from Ruby docs. RubyMine has a fantastic feature where you can Cmd + Click on Rails methods and the documentation for that method is nicely displayed above the method 😄)

![image](https://github.com/sanger/sequencescape/assets/25484917/865d5896-4ae2-4994-a0a3-dfef7540eeee)

- So, if we get rid of `alias_association`, we will have to keep `asset` terminology for the factories ([`spec/factories/plate_barcode_factories.rb`](https://github.com/sanger/sequencescape/blob/574bf5668156baee56990ccf26027229a428f78a/spec/factories/plate_barcode_factories.rb) for instance). 
```ruby
factory :plate_barcode, class: 'Barcode' do
  format { 'sequencescape22' }
  barcode { "#{configatron.plate_barcode_prefix}-#{generate(:barcode_number)}" }
  asset { build(:labware) }.   # To rename this to labware, we need alias_association
end
```
- We _can_ get rid of the `alias_association` keeping the factories as they are - i.e., _with_ the asset terminology.
- So, if we are to completely get rid of code references that refer to assets without aliasing, it is my opinion that should be followed by a migration. This way it will nicely align with what Rails expects in `ActiveRecord` as it does not allow native support for aliasing relations.
- Aliasing relations is actually a [hot topic in Rails community](https://discuss.rubyonrails.org/t/feature-proposal-alias-association/83439), as [Shopify](https://www.shopify.com/uk) developers submitted a [pretty hefty PR](https://github.com/rails/rails/pull/49801) for Rails main branch to be included in a subsequent Rails release. Spoiler alert: the PR is still open.